src/main/java/fr/unice/polytech/si3/miaou/brainfuck/instructions/Procedure.java

```diff
@@ -32,7 +54,21 @@ public Procedure(String name, int position) {
 	 */
 	@Override
 	public void accept(Machine machine) {
+		if (parameter.isPresent()) {
+			machine.saveMemoryAddress();
+			machine.setLocation(parameter.get());
+		}
 		machine.saveReturnAddress();
 		machine.setInstrPointer(position-1); //-1, because the instruction pointer is incremented right after the execution of the Procedure instruction.
 	}
+
+	/**
+	 * Clones the current procedure instruction.
+	 *
+	 * @return the cloned procedure instruction.
+	 */
+	@Override
+	public Procedure clone() {
```

Comment:
It is generally discouraged to override clone() method. Use a copy constructor instead. Also it doesn't seem to be used anywhere.